### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by swapnilkumarrai07**

### CODE REVIEW

File: DecodeCx1.py [hunk 1]
```diff
@@ -5,195 +5,160 @@
 
 
 tester_details=[]
-r=''  # To save the value of block_insights response
+n1=9  # No. of callings for all blocks
```
Comment (rajesh-entropy) [inline DecodeCx1.py]: can we rename it to `number_of_blocks` ?

File: DecodeCx1.py [hunk 2]
```diff
@@ -5,195 +5,160 @@
 
 
 tester_details=[]
-r=''  # To save the value of block_insights response
+n1=9  # No. of callings for all blocks
+n2=2  # No. of testers giving test
+n3=8  # No. of block_insights
```
Comment (rajesh-entropy) [inline DecodeCx1.py]: number of block_insights should be equal to `number_of_blocks` right?

File: DecodeCx1.py [hunk 3]
```diff
@@ -5,195 +5,160 @@
...
+login_url='https://app.dev.getdecode.io/authentication/login'
+tester_url='tester'
+get_resp_url='test/blocks?study_id=$study_id&block_id=$block_id&tester_id=$tester'
+get_resp_url_2=Template(get_resp_url)
+post_resp_url='tester_response'
+insights_url='blocks_insights'
+
+def generate_headers():
+#To generater authorization id and id token***********************************************************************************************************
+    request_json={'username': "bantupalli.sriharsha@entropiktech.com", 'password': "Harsha@10150", 'workspace': "beintern"}
```
Comment (rajesh-entropy) [inline DecodeCx1.py]: try reading password from either environment variable or arguments?

File: DecodeCx1.py [hunk 4]
```diff
@@ -5,195 +5,160 @@
...
+    c=response.content
+    c=json.loads(c.decode('utf-8'))
+    access_token=c['data']['access_token']
+    id_token=c['data']['id_token']
+    return {'authorization': access_token, 'id_token': id_token, 'workspace_id': '01GG7AW75CHDNZ2BNYKZS9ZHMF'}
+
+
+
+def generate_testers(url):
+#To generate tester id using post method*********************************************************************************************************
```
Comment (rajesh-entropy) [inline DecodeCx1.py]: maintain the indentation to this comment, and remove *'s if there is no advantage with it.

File: DecodeCx1.py [hunk 5]
```diff
@@ -5,195 +5,160 @@
...
-            request_json={"study_id":study_id,"workspace_id":headers1['workspace_id'],"preview":True,"variables":{}}
-            jsondata=json.dumps(request_json)
-            response=requests.post(url, headers=headers1, data=jsondata)
-            assert response.status_code==200, 'Error'
-            c=response.content
-            c=json.loads(c.decode('utf-8'))
-            tester=c['data']['tester']['tester_id']
-            tester_details.append(tester)
-            print()
+    url=url
```
Comment (rajesh-entropy) [inline DecodeCx1.py]: what is the use of this line?

**Review by rajesh-entropy — CHANGES_REQUESTED** [review]

File: DecodeCx1.py (hunk 2 above)
Comment (swapnilkumarrai07) [reply DecodeCx1.py]: actually no. of blocks are 9 including thankyou page. I am also calling the thankyou page.

File: DecodeCx1.py (hunk 1 above)
Comment (swapnilkumarrai07) [reply DecodeCx1.py]: yeah.. sure.

File: DecodeCx1.py (hunk 4 above)
Comment (swapnilkumarrai07) [reply DecodeCx1.py]: sure..

File: DecodeCx1.py (hunk 5 above)
Comment (swapnilkumarrai07) [reply DecodeCx1.py]: no use.. so I removed

File: DecodeCx1.py (hunk 3 above)
Comment (swapnilkumarrai07) [reply DecodeCx1.py]: yeah sure

**Pull request merged**